lib/lunar.js

```diff
@@ -163,36 +168,46 @@ function altitude_julian(h, event, t, lat, lon) {
     }
 
     if (roots === 1) {
-      if (h0 < 0) { 
-        times.rise = i + x1;
-      } else {
-        times.set = i + x1;
+      if (h0 < 0 && isNaN(times.moonrise)) {
+        times.moonrise = i + x1;
+      } else if(isNaN(times.moonset)) {
+        times.moonset = i + x1;
       }
     } else if (roots === 2) {
-      times.rise = i + (ye < 0 ? x2 : x1);
-      times.set  = i + (ye < 0 ? x1 : x2);
+      times.moonrise = isNaN(times.moonrise) ?
+          i + (ye < 0 ? x2 : x1) : times.moonrise;
+      times.moonset  = isNaN(times.moonset) ?
+          i + (ye < 0 ? x1 : x2): times.moonset;
     }
 
-    // Found what we were looking for
-    if (times[event] && times[event] < 24) {
-      return hours_later(t, times[event]);
+    // Found all the things!
+    if(!isNaN(times.moonrise) && !isNaN(times.moonset)) {
+      if(times.moonrise >= 24) {
+        times.moonrise = NaN;
+      } else {
+        times.moonrise = hours_later(t, times.moonrise);
+      }
+      if(times.moonset >= 24) {
+        times.moonrise = NaN;
+      } else {
+        times.moonset = hours_later(t, times.moonset);
+      }
```

Comment:
It feels a little fishy to me to reuse the object here. Not _bad_, just fishy. Might it make more sense to just use variables for rise and set, and then create the object when we return?